frame/staking/src/lib.rs

```diff
@@ -2368,6 +2380,76 @@ impl<T: Config> Module<T> {
 		<SnapshotNominators<T>>::kill();
 	}
 
+	/// Modify the target validator count based on the staking participation.
+	fn dynamic_damping_validator_count(
```

Comment:
Mutating state (which this function does at the last line) is an important event that should be visible to prospective authors of the code. If a function is going to have (state-wise) side effects, then the name should clearly name that. In this case, the name `dynamic_damping_validator_count ` is vaguely related but says nothing about that. 

Or, you could consider returning the new number of validators that should be set, and letting the call site `::put` it.

Reply:
>> Or, you could consider returning the new number of validators that should be set, and letting the call site ::put it.

Agree with this.